### PR TITLE
[FleetExecutor] Get global cluster information

### PR DESCRIPTION
**Pull request by LiYuRio**
### PR types

New features
### PR changes

Others
### Describe

临时从launch脚本中拿到集群信息

### CODE REVIEW

File: paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto [hunk 1]
```diff
@@ -15,7 +15,14 @@
 syntax = "proto2";
 package paddle.distributed;
 
+message RankInfo {
+  required string rank = 1;
```
Comment (FeixLiu) [inline paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto]: `required` is not recommended by Google, use `optional` with a default value?
image

Comment (FeixLiu) [reply paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto]: We can add some checks during parsing.

File: python/paddle/fluid/executor.py [hunk 2]
```diff
@@ -1848,7 +1848,19 @@ def _run_using_fleet_executor(self,
...
+        if cur_rank and rank_to_ip_port_str:
+            rank_to_ip_port = eval(rank_to_ip_port_str)
+            fleet_exe_desc.cur_rank = cur_rank
+            for rank in rank_to_ip_port:
+                rank_info = fleet_executor_desc_pb2.RankInfo()
+                rank_info.rank = str(rank)
+                rank_info.ip_port = rank_to_ip_port[rank]
+                fleet_exe_desc.cluster_info.append(rank_info)
+        else:
+            logging.info("Fleet Executor will run on single device only.")
```
Comment (FeixLiu) [inline python/paddle/fluid/executor.py]: Leave a `warning` here?

File: python/paddle/fluid/tests/unittests/test_fleet_executor.py [hunk 3]
```diff
@@ -31,6 +32,10 @@ def run_fleet_executor(self, place):
         }
         exe.run(empty_program, feed={'x': [1]})
 
+    def test_environment_variables(self):
+        self.assertIsNone(os.getenv["PADDLE_FLEET_EXECUTOR_RANK_TO_IP_PORT"])
```
Comment (FeixLiu) [inline python/paddle/fluid/tests/unittests/test_fleet_executor.py]: set env in the test

File: paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto (hunk 1 above)
Comment (LiYuRio) [reply paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto]: 这里一定是required的，因为下面是Repeat

Comment (wangxicoding) [reply paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto]: int类型？

File: paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto [hunk 4]
```diff
@@ -15,7 +15,14 @@
...
+  required string ip_port = 2;
+}
+
 message FleetExecutorDesc {
   optional string grain = 1 [ default = "coarse" ];
-  repeated string addrs = 2; // "ip:port" of all ranks
+  optional string cur_rank = 2
```
Comment (wangxicoding) [inline paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto]: int类型？ 另外nranks是不是也需要

File: python/paddle/distributed/fleet/launch.py [hunk 5]
```diff
@@ -326,6 +326,17 @@ def launch_collective(args):
...
         print("launch proc_id:{} idx:{}".format(proc.proc.pid, idx))
 
+    # Used by fleet executor only, store rank to ip:port in env.
+    # These variables will be read and stored in fleet_executor_desc.
+    # This is only a temporary solution, will be removed soon.
+    fleet_executor_rank_to_ip_port = {}
+    for pod in cluster.pods:
+        fleet_executor_rank_to_ip_port[pod.rank] = \
+                str(pod.addr) + ":" + str(pod.port)
+    os.environ["PADDLE_FLEET_EXECUTOR_RANK_TO_IP_PORT"] = str(
```
Comment (wangxicoding) [inline python/paddle/distributed/fleet/launch.py]: 从原来的那些环境变量就能获取吧

File: paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto (hunk 4 above)
Comment (LiYuRio) [reply paddle/fluid/distributed/fleet_executor/fleet_executor_desc.proto]: nranks可以通过cluster里的数目得到，要不先不加了

**Review by wangxicoding — APPROVED** [review]
LGTM

**Pull request merged**